src/mscorlib/shared/System/Threading/ExecutionContext.cs

```diff
@@ -63,15 +64,16 @@ private ExecutionContext ShallowClone(bool isFlowSuppressed)
         {
             Debug.Assert(isFlowSuppressed != m_isFlowSuppressed);
 
-            if (!isFlowSuppressed &&
-                (m_localValues == null ||
-                 m_localValues.GetType() == typeof(AsyncLocalValueMap.EmptyAsyncLocalValueMap))
-               )
+            if (m_localValues == null ||
+                m_localValues.GetType() == typeof(AsyncLocalValueMap.EmptyAsyncLocalValueMap))
             {
-                return null; // implies the default context
+                return isFlowSuppressed ?
+                    DefaultFlowSuppressed :
+                    null; // implies the default context
             }
+
             // Flow suppressing a Default context will have null values, set them to Empty
```

Comment:
Comment no longer valid?

Reply:
Yeah, thought I deleted that, I guess not.